lib/src/view/design_main_strand.dart

```diff
@@ -324,6 +334,15 @@ assigned, assign the complementary domain names sequence to this strand.
 ''',
           on_click: assign_domain_name_complement_from_bound_strands,
         ),
+        if (app.state.ui_state.select_mode_state.domains_selectable())
```

Comment:
This should not be a separate menu item. Just use the original menu item, but use the Selectable information to determine whether to assign to selected domains, or selected strands. If domains are not selectable, then exactly the behavior for `assign_domain_name_complement_from_bound_strands` should happen. If domains are selectable, then the new behavior should be implemented.

Reply:
We added a commit to implement this just now.